### PR TITLE
Add 1.10 blog post

### LINKED ISSUE #3603: 1.10 Release
January 10 will be the first anniversary of prettier, it should be a good time for a release :) Creating this issue to start the process.

I have started writing a retrospective blog post to put at the beginning of the releases note: https://github.com/prettier/prettier/releases (only people with commit access can see it for now).

### PR DESCRIPTION
**Pull request by j-f1**
Copied from releases page

Feel free to edit and add stuff.

Closes #3603.

[<sub><sub></sub></sub> Live preview](https://deploy-preview-3659--prettier.netlify.com/blog/2018/01/10/1.10.0.html)

### CODE REVIEW

File: website/blog/2018-01-10-1.10.0.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+---
+author: Christopher Chedeau (@vjeux)
+authorURL: https://twitter.com/vjeux
+title: "Prettier 1.10: One Year of Prettier 🎉"
+---
+
+Happy Prettier-versary!
+
+This release adds partial support for Vue ([#3563]), fixes the printing of Flow type annotation comments ([#3449]), and fixes a bug with comments in SCSS selectors ([#3649])
```
Comment (j-f1) [inline website/blog/2018-01-10-1.10.0.md]: I feel like these are the major changes (should the plugin API be publicized in this release?), but feel free to change this if you feel that some of these aren’t important or something important is missing.

**Comment by azz** [comment]
Ok I did everything in the release notes sans the actual descriptions. Felt like doing rote work. Anyone want to fill in the descriptions? Otherwise I'll tackle that tomorrow.

File: website/blog/2018-01-10-1.10.0.md [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+### Flow
+
+#### Print flow type annotation comments as comments ([#3449]) by [@duailibe]
+
+```ts
+// Before
+let foo: string = "a";
+
+// After
+let foo /*: string */ = "a";
```
Comment (SimenB) [inline website/blog/2018-01-10-1.10.0.md]: this example makes it seem like prettier converts normal type annotations into the comment type. (type `ts` in the code fence is sorta weird as well, but doesn't matter I suppose)

Comment (SimenB) [reply website/blog/2018-01-10-1.10.0.md]: I suppose it's the same of the other examples. Maybe add a "before prettier" variant in addition to the comparison between prettier 1.9 and 1.10?

Comment (j-f1) [reply website/blog/2018-01-10-1.10.0.md]: I don’t think there’s flow highlighting, and regular JS highlighting doesn’t highlight Flow stuff properly.

File: website/blog/2018-01-10-1.10.0.md [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+### JavaScript
+
+#### Print comment after arrow function params ([#3444]) by [@duailibe]
+
+```ts
+// Before
+const run = (cmd /*: string */ /*: Promise<void> */) => {};
+
+// After
+const run = (cmd /*: string */ /*: Promise<void> */) => {};
```
Comment (j-f1) [inline website/blog/2018-01-10-1.10.0.md]: This doesn’t look any different.

File: website/blog/2018-01-10-1.10.0.md [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+```
+
+### TypeScript
+
+### Use TypeScript to 2.7-insiders ([#3580]) by [@azz]
```
Comment (j-f1) [inline website/blog/2018-01-10-1.10.0.md]: `Update`?

File: website/blog/2018-01-10-1.10.0.md (hunk 2 above)
Comment (azz) [reply website/blog/2018-01-10-1.10.0.md]: The description bit still hasn't been written explaining that comment types will be preserved.

File: website/blog/2018-01-10-1.10.0.md [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+
+<!-- prettier-ignore -->
+`````md
+<!-- before -->
+
+> ````md
+> <!-- prettier-ignore -->
+> ```js
+ugly   ( code ) ;
+> ````
```
Comment (j-f1) [inline website/blog/2018-01-10-1.10.0.md]: Looks like the ` ``` ` is missing here.

File: website/blog/2018-01-10-1.10.0.md [hunk 6]
```diff
@@ -0,0 +1,381 @@
...
+
+#### Add `options` field to `getSupportInfo()` ([#3433]) by [@ikatyang]
+
+#### Only search for `.editorconfig` up to the VCS directory ([#3559]) by [@josephfrazier]
+
+### CLI
+
+#### Make all CLI logging go through a logger ([#3515]) by [@azz]
+
+```bash
```
Comment (j-f1) [inline website/blog/2018-01-10-1.10.0.md]: I’ve found that ` ```shellsession ` looks better for shell sessions like this.

**Comment by azz** [comment]
I accidentally formatted the file with prettier before I added the `<-- prettier-ignore -->` to all the code blocks 😢

File: website/blog/2018-01-10-1.10.0.md (hunk 6 above)
Comment (azz) [reply website/blog/2018-01-10-1.10.0.md]: Ooh nice 🏆

File: website/blog/2018-01-10-1.10.0.md [hunk 7]
```diff
@@ -401,4 +401,4 @@ We've fixed up the behavior of the CLI when it comes to ignored files. Previousl
 [#3641]: https://github.com/prettier/prettier/pull/3641
 [#3643]: https://github.com/prettier/prettier/pull/3643
 [#3649]: https://github.com/prettier/prettier/pull/3649
-[plugin api]: https://prettier.io/docs/en/plugins
```
Comment (azz) [inline website/blog/2018-01-10-1.10.0.md]: it works without `.html`

File: website/blog/2018-01-10-1.10.0.md [hunk 8]
```diff
@@ -380,7 +380,7 @@ We've fixed up the behavior of the CLI when it comes to ignored files. Previousl
 [@j-f1]: https://github.com/j-f1
 [@thorn0]: https://github.com/thorn0
 [@vjeux]: https://github.com/vjeux
-[#2097]: https://github.com/prettier/prettier/issue/2097
+[#2097]: https://github.com/prettier/prettier/issues/2097
```
Comment (azz) [inline website/blog/2018-01-10-1.10.0.md]: Oh that's annoyingly inconsistent... It's `/pulls` for the PR list, `/pull/:number` for a PR, then `/issues` for the issues list, and `/issues/:number` for an issue.

File: website/blog/2018-01-10-1.10.0.md (hunk 7 above)
Comment (ikatyang) [reply website/blog/2018-01-10-1.10.0.md]: Yeah, I've noticed that, but I think `.html` should be better as it's a static site(?).

Comment (azz) [reply website/blog/2018-01-10-1.10.0.md]: 🤷‍♂️

File: website/blog/2018-01-10-1.10.0.md [hunk 9]
```diff
@@ -0,0 +1,404 @@
+---
+author: Christopher Chedeau (@vjeux)
+authorURL: https://twitter.com/vjeux
```
Comment (vjeux) [inline website/blog/2018-01-10-1.10.0.md]: It's unfortunate, I only wrote half of it...

File: website/blog/2018-01-10-1.10.0.md [hunk 10]
```diff
@@ -0,0 +1,404 @@
...
+
+There are three official plugins under development:
+
+* [`@prettier/plugin-python`](https://github.com/prettier/prettier-python)
+* [`@prettier/plugin-php`](https://github.com/prettier/prettier-php)
+* [`@prettier/plugin-swift`](https://github.com/prettier/prettier-swift)
+
+All three of these plugins are still under active development and aren't _quite_ really to throw production code at, but keep an eye on them as they will be soon! If you're interested in helping build those plugins, check out their repository issues list or run it against your own code and start raising bugs! Similarly, if you're interested in building a new language in Prettier and need some help getting it off the ground, [create an issue in the prettier repo](https://github.com/prettier/prettier/issues/new) and let us help you out.
+
+Prettier's in-built languages have been refactored to be expressed in terms of the plugin API, this way we can guarantee going forward that the core API going forward is generic.
```
Comment (pbarbiero) [inline website/blog/2018-01-10-1.10.0.md]: > going forward that the core API going forward  

redundancy

Comment (azz) [reply website/blog/2018-01-10-1.10.0.md]: Thanks 😄

File: website/blog/2018-01-10-1.10.0.md [hunk 11]
```diff
@@ -0,0 +1,411 @@
...
+
+Check [the docs][plugin api] for more information.
+
+> Please note that as this is a new, significantly large feature, we're releasing this under a **beta** label in the documentation. While we don't anticipate anything significant, this means that we may make breaking changes in the next release.
+
+### Flow
+
+#### Print flow type annotation comments as comments ([#3449]) by [@duailibe]
+
+Flow [type annotation comments](https://flow.org/en/docs/types/comments/#toc-comment-type-annotation) are a nice way to get type checking without having to transpile your files. But before this release, Prettier would not print the type annotation as comments, making it impossible to use that feature; now Prettier will correctly detect if the type annotation was a comment and print accordingly.
```
Comment (vjeux) [inline website/blog/2018-01-10-1.10.0.md]: Not that it doens't work yet for `/*=` ones and this only works for the `flow`. In Babylon we also improved the position of various comments. cc @duailibe. So this is one step in the right direction and people should give us more examples where it's not working so we can fix it.

File: website/blog/2018-01-10-1.10.0.md [hunk 12]
```diff
@@ -0,0 +1,411 @@
...
+    // comment
+    () =>
+      true
+  }
+/>;
+```
+
+#### Don't wrap JSX elements as attribute in `()` ([#3640]) by [@duailibe]
+
+The [JSX spec](http://facebook.github.io/jsx/) allows passing an element as an attribute to another element and Prettier was adding parens around, causing a syntax error. This was fixed in this release.
```
Comment (vjeux) [inline website/blog/2018-01-10-1.10.0.md]: You can mention that even though the jsx spec allows it, no current transpiler actually prints it correctly. Babel will with the v7

File: website/blog/2018-01-10-1.10.0.md (hunk 11 above)
Comment (duailibe) [reply website/blog/2018-01-10-1.10.0.md]: @vjeux I'm aware #3493 :-) I thought about adding a "request for help" or something

File: website/blog/2018-01-10-1.10.0.md [hunk 13]
```diff
@@ -0,0 +1,411 @@
...
+
+Using plugins is as simple as installing them via `npm`/`yarn`, and running Prettier as you usually would, there's no additional configuration to worry about!
+
+There are three official plugins under development:
+
+* [`@prettier/plugin-python`](https://github.com/prettier/prettier-python)
+* [`@prettier/plugin-php`](https://github.com/prettier/prettier-php)
+* [`@prettier/plugin-swift`](https://github.com/prettier/prettier-swift)
+
+All three of these plugins are still under active development and aren't _quite_ really to throw production code at, but keep an eye on them as they will be soon! If you're interested in helping build those plugins, check out their repository issues list or run it against your own code and start raising bugs! Similarly, if you're interested in building a new language in Prettier and need some help getting it off the ground, [create an issue in the prettier repo](https://github.com/prettier/prettier/issues/new) and let us help you out.
```
Comment (lipis) [inline website/blog/2018-01-10-1.10.0.md]: `prettier` -> `Prettier`

Comment (azz) [reply website/blog/2018-01-10-1.10.0.md]: I meant to use lower case here to denote the repository.

**Comment by azz** [comment]
@vjeux Do you want to fill out the graphql section?

File: website/blog/2018-01-10-1.10.0.md [hunk 14]
```diff
@@ -0,0 +1,445 @@
...
+// Before
+const run = (cmd /*: string */ /*: Promise<void> */) => {};
+
+// After
+const run = (cmd /*: string */) /*: Promise<void> */ => {};
+```
+
+### TypeScript
+
+### Support numeric separators ([#3580]) by [@azz]
```
Comment (MichaelDeBoey) [inline website/blog/2018-01-10-1.10.0.md]: This should have 4 #'s I think 🙂

Comment (j-f1) [reply website/blog/2018-01-10-1.10.0.md]: Thanks for pointing this out!

File: website/blog/2018-01-10-1.10.0.md [hunk 15]
```diff
@@ -0,0 +1,444 @@
...
+
+Using plugins is as simple as installing them via `npm`/`yarn`, and running Prettier as you usually would, there's no additional configuration to worry about!
+
+There are three official plugins under development:
+
+* [`@prettier/plugin-python`](https://github.com/prettier/prettier-python)
+* [`@prettier/plugin-php`](https://github.com/prettier/prettier-php)
+* [`@prettier/plugin-swift`](https://github.com/prettier/prettier-swift)
+
+All three of these plugins are still under active development and aren't _quite_ really to throw production code at, but keep an eye on them as they will be soon! If you're interested in helping build those plugins, check out their repository issues list or run it against your own code and start raising bugs! Similarly, if you're interested in building a new language in Prettier and need some help getting it off the ground, [create an issue in the prettier repo](https://github.com/prettier/prettier/issues/new) and let us help you out.
```
Comment (existentialism) [inline website/blog/2018-01-10-1.10.0.md]: > All three of these plugins are still under active development and aren't _quite_ really to throw production code at, but keep an eye on them as they will be soon! 

Maybe:

> All three of these plugins are still under active development and not _quite_ **ready** for production code yet, but will be soon!

File: website/blog/2018-01-10-1.10.0.md [hunk 16]
```diff
@@ -244,6 +244,8 @@ The [JSX spec](http://facebook.github.io/jsx/) allows passing an element as an a
 
 #### Print comments inside of selector as is ([#3649]) by [@vjeux]
 
+Unfortunately the CSS parser we're using for selectors does not support comments. As such there was a case where comments where selectors could be moved onto lines with comments. This has now been fixed.
```
Comment (suchipi) [inline website/blog/2018-01-10-1.10.0.md]: This only applies to line comments (`//`), not block comments (`/* */`)

**Pull request merged**